feat: 듀오 찾기 수정

for2ver0 opened the pull request:
## ✨ 담당 파트
- 듀오 찾기 수정
<br>

## 🔎 작업 상세 내용
### 1. Duo entity 수정
- 듀오 찾기 수정에 따른 Duo 엔티티 수정
- DB의 예약어인 rank 를 ranks 로 수정

### 2. DuoUpdateRequest 추가
- 듀오 찾기 수정의 Request DTO

### 3. DuoController 수정
- 듀오 찾기 수정 메서드 update() 추가

### 4. Kda entity 에 private 생성자 및 정적 팩토리 메서드 추가
- private 생성자
- 정적 팩토리 메서드 of

### 5. Riot API 를 이용하여, 반환받는 전적 기록 RecordResponse 추가
- wins: 승리 횟수
- losses: 패배 횟수
- kills: 총 kill 횟수
- deaths: 총 death 횟수
- assists: 총 assist 횟수
- playCounts: 게임 플레이 횟수
- kda: 각각의 평균 K/D/A를 담고 있는 Kda

### 6. DuoUpdateResponse 추가
- 듀오 찾기 수정 후 반환하는 DTO
<br>

## 🔧 앞으로의 과제
- 당장 추가될 부분은 코드의 todo에 적어놨습니다

## ✅ 테스트 코드 작성 및 기능 테스트 여부
- [ ] 테스트 코드 작성
- [ ] 기능 테스트 여부
<br>

## ➕ 이슈 링크
- #13

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- 듀오 찾기 수정에 따른 Duo 엔티티 수정
- DB의 예약어인 rank 를 ranks 로 수정

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- 듀오 찾기 수정의 Request DTO 인 DuoUpdateRequest

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- DuoController 에 듀오 찾기 수정 메서드 update() 추가

hubaek commented on src/main/java/com/summoner/lolhaeduo/domain/duo/dto/DuoUpdateRequest.java:
```diff
@@ -0,0 +1,56 @@
+    // 큐 타입에 따른 유효성 검사
+    @AssertTrue(message = "큐 타입이 빠른 대전인 경우, primaryChamp 와 secondaryRole, secondaryChamp 는 NULL 이어야 합니다.")
+    public boolean isFlexQueueTypeValid() {
+        if (queueType == QueueType.FLEX) {
```
!= QUICK 일때 조건으로 하시면 더 깔끔 할 것 같습니다!
현재는 SOLO와 FLEX 두개의 조건이 필요해요!

yokxim2 commented on src/main/java/com/summoner/lolhaeduo/domain/duo/service/DuoService.java:
```diff
@@ -124,4 +135,141 @@ private LeagueEntryResponse getSelectedRankInfoByQueueType(List<LeagueEntryRespo
+//        if (!duoLinkedAccount.getMemberId().equals(memberId)) {
+//            throw new IllegalStateException("듀오 신청자의 롤 계정이 아닙니다.");
+//        }
+//        // 중복되는 검증 로직인 듯 합니다.
```
이거 주석 없애주셔도 될거 같습니다~

yokxim2 commented on src/main/java/com/summoner/lolhaeduo/domain/duo/service/DuoService.java:
```diff
@@ -124,4 +135,141 @@ private LeagueEntryResponse getSelectedRankInfoByQueueType(List<LeagueEntryRespo
+        log.info("30일 전 Epoch Time {}", epochTime);
+
+        return epochTime;
+    }
```
이거는 너무 모양이 한달 전 시간만 계산 가능한 계산기인거 같아요.
사실 이 메서드가 서비스 로직에 들어가야 하는지도 좀 모르겠지만.. 재사용성을 위해서 인수에 몇일 전 시간 계산인지 인자를 추가해주시겠어요?

```
public long calculateStartTime(int days) {
        // 현재 시간
        Instant now = Instant.now();

        // 시간 계산
        Instant thirtyDaysAgo = now.minus(days, ChronoUnit.DAYS);

        // Epoch Time 으로 변환
        long epochTime = thirtyDaysAgo.getEpochSecond();
        log.info(days + "일 전 Epoch Time {}", epochTime);

        return epochTime;
    }
```

yokxim2 reviewed (approved):
좋아요~ 자잘한거 몇개만 수정 해주세요!

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- Kda entity 에 정적 팩토리 메서트 추가

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- Riot API 를 이용하여, 반환받는 전적 기록 RecordResponse
  - wins: 승리 횟수
  - losses: 패배 횟수
  - kills: 총 kill 횟수
  - deaths: 총 death 횟수
  - assists: 총 assist 횟수
  - playCounts: 게임 플레이 횟수
  - kda: 각각의 평균 K/D/A를 담고 있는 Kda

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- 듀오 찾기 수정 후 반환하는 DTO 인 DuoUpdateResponse

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- DuoUpdateRequest 유효성 검사 메서드 테스트

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- DuoService 에 듀오 찾기 수정 메서드 update() 추가

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- 불필요한 주석 삭제
- calculateStartTime 메서드 리팩토링

for2ver0 committed:
feat(duo-update): 듀오 찾기 수정 #13

- DuoUpdateRequest 의 유효성 검사 메서드의 잘못된 조건 수정
